pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/LimitZeroSelectionOperator.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.query;
+
+import com.linkedin.pinot.common.request.Selection;
+import com.linkedin.pinot.common.utils.DataSchema;
+import com.linkedin.pinot.core.indexsegment.IndexSegment;
+import com.linkedin.pinot.core.operator.BaseOperator;
+import com.linkedin.pinot.core.operator.ExecutionStatistics;
+import com.linkedin.pinot.core.operator.blocks.IntermediateResultsBlock;
+import com.linkedin.pinot.core.query.selection.SelectionOperatorUtils;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+
+public class LimitZeroSelectionOperator extends BaseOperator<IntermediateResultsBlock> {
```

Comment:
Trying to understand, why do we need a separate operator for this? Seems too much work for a special case. Shouldn't the existing operators take care of this case?

Reply:
Also, name could be more generic like EmptySelectionOperator?

Reply:
This operator is specific for selection with limit less or equal to 0. We don't need to fetch any data, so we don't need to create an underlying operator. The branch logic happens in the SelectionPlanNode, and with a separate operator, the logic is more clear.
I agree on the name EmptySelectionOperator.